github/retry.go

```diff
@@ -29,28 +24,15 @@ func (t *retryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	var err error
 	retry(func() error {
 		r, err = t.T.RoundTrip(req)
-		if err != nil {
-			if err == context.Canceled {
-				return &errUnretriable{Err: err}
-			}
-
-			return err
-		}
-
-		if r.StatusCode == http.StatusOK {
+		if err == context.Canceled {
 			return nil
 		}
 
-		body, _ := ioutil.ReadAll(r.Body)
-
-		// Restore the io.ReadCloser
-		r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
-
-		err = fmt.Errorf("non-200 OK status code: %v body: %q", r.Status, body)
-		if r.StatusCode > 500 {
+		if err != nil || r.StatusCode >= 500 {
 			return err
 		}
-		return &errUnretriable{Err: err}
+
+		return nil
```

Comment:
why do you return `nil`? with status code 4xx it must return error.

Reply:
ah. the error is actually returned correctly as it is defined outside. wuuuh. Why did you have to change it? The code looks more or less the same as before but much for confusing.

Reply:
imo now is clearer and cleaner.
now, the `retry` function only retries if the function to eval returns an error, and skip it if it returned no error.

Reply:
at least I failed to understand it. Also it's different from the most common retry interface https://godoc.org/github.com/cenkalti/backoff#Retry

Reply:
Current code is clearer than the older one imo; but yes, it's also better to consider the alternative of discarding both, and use `Backof` instead (which is also standard as pointed by @se7entyse7en [suggestion](https://github.com/src-d/metadata-retrieval/pull/57#discussion_r334973716))
I will commit separately.

Reply:
> Current code is clearer than the older one imo

it's cleaner for you because you wrote it. But 2 people (hopefully not the dumbest ones) failed to understand it. It does mean something :)

Reply:
Current code in `master`:
- `retry` returns either an error, an unwrapped error or `nil` unnecessarily because the output of `retry` is not used at all,
- is manipulating the error returned by the inner `t.T.RoundTrip`, changing it with a handmade `non-200 something` error, which is not the purpose of the `RetryTransport`, that should, imo, just retry or skip.

Anyhow, we're also getting rid of that now that we're going to use `Backof` per @se7entyse7en [suggestion](https://github.com/src-d/metadata-retrieval/pull/57#discussion_r334973716)